pkgs/applications/editors/emacs/elisp-packages/manual-packages/lsp-bridge/default.nix

```diff
@@ -0,0 +1,78 @@
+{ lib
+, trivialBuild
+, fetchFromGitHub
+, python310Packages
+, python310
+, posframe
+, markdown-mode
+, yasnippet
+, org
+, which-key
+, makeWrapper
+}:
+
+let
+  rev = "7dfeeb640d14697755e2ac7997af0ec6c413197f";
+in trivialBuild {
+  pname = "lsp-bridge";
+  version = "20230104";
+
+  commit = rev;
+
+  src = fetchFromGitHub {
+    owner = "manateelazycat";
+    repo = "lsp-bridge";
+    inherit rev;
+    sha256 = "sha256-sB5niigN0rdtqeprlZAJEKgAuQDkcUMbbL9yTnrdoLg=";
+  };
+
+  patches = [ ./wrapper.patch ];
+
+  packageRequires =
+    [
+      posframe
+      markdown-mode
+      yasnippet
+      org
+      which-key
+    ];
+
+  buildInputs =
+    [
+      (python310.withPackages (ps: with ps; [ epc orjson sexpdata six ]))
+      makeWrapper
+    ];
+
+  buildPhase = ''
+    runHook preInstall
+
+    install -d $out/share/emacs/site-lisp/
+    install *.el $out/share/emacs/site-lisp/
+    install acm/*.el $out/share/emacs/site-lisp/
+    install *.py $out/share/emacs/site-lisp/
+    cp -r core $out/share/emacs/site-lisp/
+    cp -r langserver $out/share/emacs/site-lisp/
+    cp -r multiserver $out/share/emacs/site-lisp/
+    cp -r resources $out/share/emacs/site-lisp/
+    cp -r acm/icons $out/share/emacs/site-lisp/
+
+    runHook postInstall
+  '';
+
+  postInstall = with python310Packages; ''
+          wrapProgram $out/share/emacs/site-lisp/lsp_bridge.py \
+              --prefix PYTHONPATH : "${python310}" \
+              --prefix PYTHONPATH : "${epc}" \
+              --prefix PYTHONPATH : "${orjson}" \
+              --prefix PYTHONPATH : "${sexpdata}" \
+              --prefix PYTHONPATH : "${six}";
```

Comment:
This is not going to fly.

`PYTHONPATH` leaks dependencies via env vars which also leaks into other processes launched by emacs.

It's better to use a `withPackages` closure as that is more hermetic.

Reply:
Understood, didn't knew that. My first attempt to solve this, before using this wrapper, was to use propagatedBuildInputs. But it doesn't pull in these python packages as runtime dependeny, even when using

`
(python3.withPackages (ps: with ps; [ epc orjson sexpdata six ]))
`

Do you have a tip for me or could me point in the right direction?

Reply:
> Understood, didn't knew that.

Very few people do, it's one of really sharp edges integrating Nix/Python. 

This is totally untested, but something along these lines:
``` nix
let
  python = python3.withPackages (ps: with ps; [ epc orjson sexpdata six ]);
in
stdenv.mkDerivation {
  postPatch = ''
    substituteInPlace lsp-bridge.el --replace '(defcustom lsp-bridge-python-command (if (memq system-type '(cygwin windows-nt ms-dos)) "python.exe" "python3")' '(defcustom lsp-bridge-python-command "${python.interpreter}")'
  '';
}
```

That way lsp-bridge simply executes the wrapped interpreter and then we don't have to deal with `PYTHONPATH` at all :)

Reply:
This works like a charm. Thanks for your great support! :)